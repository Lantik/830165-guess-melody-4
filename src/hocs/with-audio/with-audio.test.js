import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import withAudio from './with-audio.js';

const MockComponent = (props) => {
  return (<div>
    { props.children }
  </div>);
};

MockComponent.propTypes = {
  children: PropTypes.node.isRequired
};

const MockComponentWithAudio = withAudio(MockComponent);

it(`withAudio hoc rendered correctly`, () => {
  const tree = renderer
    .create(<MockComponentWithAudio
      src={``}
      isPlaying={true}
      onPlayButtonClick={() => {}}
    />, {
      createNodeMock: () => {
        return {};
      }
    }).toJSON();

  expect(tree).toMatchSnapshot();
});
