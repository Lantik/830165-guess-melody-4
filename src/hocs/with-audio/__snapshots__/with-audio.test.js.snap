// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAudio hoc rendered correctly 1`] = `
<div>
  <audio />
</div>
`;
